src/components/Projects/WBS/wbs.jsx

```diff
@@ -11,14 +11,55 @@ import { NavItem } from 'reactstrap';
 import AddWBS from './AddWBS';
 import WBSItem from './WBSItem/WBSItem';
 import { boxStyle } from 'styles';
+import './wbs.css';
 
 const WBS = props => {
+  const { allProjects } = props;
   const projectId = props.match.params.projectId;
+  const [sortBy, setSortBy] = useState('default');
+  const [clickCount, setClickCount] = useState(0);
+  const [sortedItems, setSortedItems] = useState([]);
+
+  const projectName = allProjects.find(project => project._id === projectId).projectName;
 
   useEffect(() => {
     props.fetchAllWBS(projectId);
   }, [projectId]);
 
+  useEffect(() => {
+    // Sort WBS items on initial load if needed
+    if (sortBy !== 'default') {
+      setSortedItems(sortWBSItems());
+    } else {
+      // If default sort order, maintain the order in which items were created
+      setSortedItems([...props.state.wbs.WBSItems]);
+    }
+  }, [props.state.wbs.WBSItems, sortBy]);
+
+  const toggleSort = () => {
+    let newSortBy;
+    if (clickCount === 0) {
+      newSortBy = 'asc';
+    } else if (clickCount === 1) {
+      newSortBy = 'desc';
+    } else {
+      newSortBy = 'default';
+    }
+    setSortBy(newSortBy);
+    setClickCount((clickCount + 1) % 3); // Cycle through 0, 1, 2
+  };
```

Comment:
I don't think implementing toggleSort by keeping track of click counts is a good approach. The more useState we put into the component, the worse the performance will be.